File: pinyougou-parent/pinyougou-sellergoods-interface/src/main/java/com/pinyougou/sellergoods/service/ItemCatService.java
package com.pinyougou.sellergoods.service;
import java.util.List;
import com.pinyougou.pojo.TbItemCat;

/**
 * 
 * Title:ItemCatService.java
 * Description: 商品分类服务接口
 * @author xll
 * @date 2019年2月8日 下午2:48:29
 * @version 1.0
 *
 */
public interface ItemCatService {

	 /**
	  * Title:findByParentId
	  * Description: 根据父级id查询商品分类
	  * @param parentId
	  * @return
	  */
	 public List<TbItemCat> findByParentId(Long parentId);
	 
	 /**
	  * Title:add
	  * Description: 添加商品分类
	  * @param itemCat
	  */
	 public void add(TbItemCat itemCat);
	 
	 /**
	  * Title:findOne
	  * Description: 根据Id查询商品分类实例
	  * @param id
	  * @return
	  */
	 public TbItemCat findOne(Long id);
	 
	 /**
	  * Title:update
	  * Description:修改商品分类信息
	  * @param itemCat
	  */
	 public void update(TbItemCat itemCat);
	 
	 /**
	  * Title:delete
	  * Description:判断能否删除 能则删除返回true 否则不删除返回false
	  * @param ids
	  * @return
	  */
	 public boolean delete(Long[] ids);
	 
	 /**
	  * Title:findAll
	  * Description: 查询所有商品分类
	  * @return
	  */
	 public List<TbItemCat> findAll();
}

File: pinyougou-parent/pinyougou-manager-controller/src/main/webapp/js/service/itemCatService.js
//服务层
app.service('itemCatService',function($http){
	    	
	//根据父级id查询商品分类列表
	this.findByParentId = function(parentId){
		return  $http.get('../itemCat/findByParentId.do?parentId=' + parentId);
	}
	
	//添加商品类型
	this.add = function(entity){
		return $http.post('../itemCat/add.do',entity);
	}
	
	//查询商品类型实例
	this.findOne = function(id){
		return $http.get('../itemCat/findOne.do?id=' + id);
	}
	
	//修改商品分类信息
	this.update = function(entity){
		return $http.post('../itemCat/update.do',entity);
	}
	
	//批量删除商品分类
	this.dele = function(ids){
		return $http.get('../itemCat/delete.do?ids=' + ids);
	}
	
	//查询全部商品
	this.findAll = function(){
		return $http.get('../itemCat/findAll.do');
	}
});
